tools/chapel-py/src/core-types.cpp

```diff
@@ -537,12 +647,60 @@ PyObject* wrapAstNode(ContextObject* context, const AstNode* node) {
 #undef AST_LEAF
 #undef AST_BEGIN_SUBCLASSES
 #undef AST_END_SUBCLASSES
+#undef CAST_TO
     default: break;
   }
   Py_XDECREF(args);
   return toReturn;
 }
 
+PyObject* wrapType(ContextObject* context, const types::Type* node) {
+  PyObject* toReturn = nullptr;
+  if (node == nullptr) {
+    Py_RETURN_NONE;
+  }
+  PyObject* args = Py_BuildValue("(O)", (PyObject*) context);
+  switch (node->tag()) {
+#define CAST_TO(NAME) \
+    case types::typetags::NAME: \
+      toReturn = PyObject_CallObject((PyObject*) &NAME##Type, args); \
+      ((NAME##Object*) toReturn)->parent.ptr = (const types::Type*) node->to##NAME(); \
+      break;
+#define TYPE_NODE(NAME) CAST_TO(NAME)
+#define BUILTIN_TYPE_NODE(NAME, CHPL_NAME) CAST_TO(NAME)
+#define TYPE_BEGIN_SUBCLASSES(NAME) /* No need to handle abstract parent classes. */
```

Comment:
I would put the comment not as part of the macro definition. Not sure that it matters to the compiler, but to a human it looks weird